examples/Cassie/run_constrained_lqr_controller_sim.cc

```diff
@@ -0,0 +1,472 @@
+DEFINE_double(contact_radius, 2e-4,
+              "The characteristic scale of contact patch (m)");
+DEFINE_string(simulation_type, "compliant",
+              "The type of simulation to use: "
+              "'compliant' or 'timestepping'");
+DEFINE_double(dt, 1e-3,
+              "The step size to use for "
+              "'simulation_type=timestepping' (ignored for "
+              "'simulation_type=compliant'");
+DEFINE_double(publish_rate, 1000, "Publishing frequency (Hz)");
+
+// Cassie model paramter
+DEFINE_bool(floating_base, false, "Fixed or floating base model");
+
+// LCM channels
+DEFINE_string(state_channel, "CASSIE_STATE",
+              "LCM channel for receiving the state");
+DEFINE_string(input_channel, "CASSIE_INPUT",
+              "LCM channel for receiving the motor inputs");
+
+/*
+ * Function to run collisionDetect and return a ContactInfo object for the four
+ * contact points in Cassie.
+ * @param tree The RigidBodyTree containing the cassie model and a ground plane
+ * box.
+ * @param q0 The Cassie position configuration for which contact information is
+ * to be computed.
+ */
+ContactInfo ComputeCassieContactInfo(const RigidBodyTree<double>& tree,
+                                     VectorXd q0) {
+  VectorXd phi_total;
+  Matrix3Xd normal_total, xA_total, xB_total;
+  vector<int> idxA_total, idxB_total;
+  KinematicsCache<double> k_cache = tree.doKinematics(q0);
+
+  // The full collisionDetect solution.
+  const_cast<RigidBodyTree<double>&>(tree).collisionDetect(
+      k_cache, phi_total, normal_total, xA_total, xB_total, idxA_total,
+      idxB_total);
+
+  const int world_ind = GetBodyIndexFromName(tree, "world");
+  const int toe_left_ind = GetBodyIndexFromName(tree, "toe_left");
+  const int toe_right_ind = GetBodyIndexFromName(tree, "toe_right");
+
+  // Extracting information into the four contacts.
+  VectorXd phi(4);
+  Matrix3Xd normal(3, 4), xA(3, 4), xB(3, 4);
+  vector<int> idxA(4), idxB(4);
+
+  int k = 0;
+  for (unsigned i = 0; i < idxA_total.size(); ++i) {
+    int ind_a = idxA_total.at(i);
+    int ind_b = idxB_total.at(i);
+    if ((ind_a == world_ind && ind_b == toe_left_ind) ||
+        (ind_a == world_ind && ind_b == toe_right_ind) ||
+        (ind_a == toe_left_ind && ind_b == world_ind) ||
+        (ind_a == toe_right_ind && ind_b == world_ind)) {
+      xA.col(k) = xA_total.col(i);
+      xB.col(k) = xB_total.col(i);
+      idxA.at(k) = idxA_total.at(i);
+      idxB.at(k) = idxB_total.at(i);
+      ++k;
+    }
+  }
+
+  ContactInfo contact_info = {xB, idxB};
+  return contact_info;
+}
+
+/*
+ * A class to serve as the input to the ConstrainedLQRController.
+ * It takes in a basic vector (of size num_states) at its input and has an
+ * output port of type OutputVector.
+ * It serves as an interfacing connector between the RigidBodyPlant's state
+ * output port and ConstrainedLQRController's info input port.
+ */
+class StateConnector : public LeafSystem<double> {
```

Comment:
It feels like this should be moved to `multibody` and given a more appropriate name. It's acting like a state estimator from a simulated plant, but not currently copying over the efforts. I think it's OK, for now, to avoid adding in the effort logic (which will require an additional input port), but others would have use for a block like this.